### PR TITLE
[Metrics SDK] Add unit to Instrument selection criteria 

### LINKED ISSUE #1995: [Metrics SDK] Add unit to Instrument selection criteria
As per the specs changes - https://github.com/open-telemetry/opentelemetry-specification/pull/3184, Unit should used in Instrument selection criteria defined in View API.

Changes would be potentially required here - https://github.com/open-telemetry/opentelemetry-cpp/blob/main/sdk/include/opentelemetry/sdk/metrics/view/meter_selector.h

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #1995 

This is breaking change for View/ViewFactory API:

Old: `View(name, desc, agg_type, agg_config, attribute_processor)`
New: `View(name, desc, unit, agg_type, agg_config, attribute_processor)`

Old: `ViewFactory::Create(name, desc, agg_type, agg_config, attribute_processor)`
New: `ViewFactory::Create(name, desc, unit, agg_type, agg_config, attribute_processor)`

Although the addition of the `unit` as an optional last argument can prevent any breaking changes, I just felt it would be more logical to include it alongside the `name` and `desc` arguments.

## Changes

Add unit to instrument selection criteria.
For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

**Review by marcalff — COMMENTED** [review]
LGTM.

Please add a CHANGELOG entry, similar to this:

```
Breaking changes:

* [SDK] MeterProvider should own MeterContext, not share it
  [#2218](https://github.com/open-telemetry/opentelemetry-cpp/pull/2218)
  * The `MeterProvider` constructor now takes a `unique_ptr` on
    `MeterContext`, instead of a `shared_ptr`.
  * Please adjust SDK configuration code accordingly.
```

**Review by marcalff — APPROVED** [review]

**Comment by marcalff** [comment]
> Although the addition of the `unit` as an optional last argument can prevent any breaking changes, I just felt it would be more logical to include it alongside the `name` and `desc` arguments.

Agreed, looks better this way.

File: sdk/include/opentelemetry/sdk/metrics/view/instrument_selector.h [hunk 1]
```diff
@@ -19,19 +19,24 @@ class InstrumentSelector
 {
 public:
   InstrumentSelector(opentelemetry::sdk::metrics::InstrumentType instrument_type,
-                     opentelemetry::nostd::string_view name)
+                     opentelemetry::nostd::string_view name,
+                     opentelemetry::nostd::string_view units)
       : name_filter_{PredicateFactory::GetPredicate(name, PredicateType::kPattern)},
+        unit_filter_{PredicateFactory::GetPredicate(units, PredicateType::kExact)},
         instrument_type_{instrument_type}
+
```
Comment (esigo) [inline sdk/include/opentelemetry/sdk/metrics/view/instrument_selector.h:27]:
```suggestion
```

**Review by esigo — APPROVED** [review]
LGTM
Thanks :)

**Comment by marcalff** [comment]
@lalitb 

Please add a CHANGELOG entry.

**Comment by lalitb** [comment]
> @lalitb
> 
> Please add a CHANGELOG entry.

Done now. Thanks.

**Pull request merged**